From 5f88f766cbf75d476787e76cd70ad2981b49af7d Mon Sep 17 00:00:00 2001
From: Randy Barlow <randy@electronsweatshop.com>
Date: Fri, 11 Mar 2016 15:29:32 -0500
Subject: [PATCH] Allow the Celery process to resolve domain names.

Fedora Rawhide restricts access to /etc/resolv.conf. This commit
adds the sysnet_dns_name_resolve policy to the celery_t context
which allows it to resolve hostnames.

https://pulp.plan.io/issues/1764

fixes #1764
---
 server/selinux/server/pulp-celery.te | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/server/selinux/server/pulp-celery.te b/server/selinux/server/pulp-celery.te
index 770111e..586cc0e 100644
--- a/server/selinux/server/pulp-celery.te
+++ b/server/selinux/server/pulp-celery.te
@@ -71,9 +71,7 @@ ifndef(`distro_rhel6', `
 # pulp_ostree sync call on F23 calls out to gpg
 # temporary workaround for https://pulp.plan.io/issues/1731
 
-ifdef(`distro_fedora23', `
-    allow celery_t self:process { setrlimit };
-')
+allow celery_t self:process { setrlimit };
 
 ######################################
 
@@ -122,6 +120,7 @@ sysnet_read_config(celery_t)
 ######################################
 #
 # Pulp workers do network related things including:
+#  - perform DNS queries
 #  - connect to mongod and amqp. This is required for correct processing of tasks
 #  - sync data to/from remote hosts via HTTP/HTTPS
 #  - send e-mail using SMTP for notification purposes
@@ -132,6 +131,7 @@ sysnet_read_config(celery_t)
 # additional protection.
 #
 
+auth_use_nsswitch(celery_t)
 allow celery_t self:tcp_socket create_stream_socket_perms;
 corenet_tcp_connect_all_ports(celery_t)
 corenet_tcp_bind_all_ports(celery_t)
-- 
2.7.2

